pkg/controllers/management/clusterdeploy/clusterdeploy.go

```diff
@@ -606,6 +613,42 @@ func (cd *clusterDeploy) getControlPlaneTaints(name string) ([]corev1.Taint, err
 	return allTaints, nil
 }
 
+// getTolerations returns a list of tolerations that combines tolerations based off control plane taints and any
+// user defined tolerations. Adding tolerations that match cp taints will allow the cluster agent to deploy onto
+// the controlplane node.
+func getTolerations(cluster *apimgmtv3.Cluster, controlPlaneTaints []corev1.Taint) []corev1.Toleration {
+	var tolerations []corev1.Toleration
+	for _, taint := range controlPlaneTaints {
+		tolerations = append(tolerations, corev1.Toleration{
+			Key:               taint.Key,
+			Value:             taint.Value,
+			Effect:            taint.Effect,
+		})
+	}
+
+	if cluster.Spec.ClusterAgentDeploymentCustomization.AppendTolerations != nil {
+		tolerations = append(tolerations, cluster.Spec.ClusterAgentDeploymentCustomization.AppendTolerations...)
+	}
+
+	return tolerations
+}
+
+func getAffinity(cluster *apimgmtv3.Cluster) *corev1.Affinity {
+	if cluster.Spec.ClusterAgentDeploymentCustomization.OverrideAffinity != nil {
+		return cluster.Spec.ClusterAgentDeploymentCustomization.OverrideAffinity
+	}
+
+	return settings.GetClusterAgentDefaultAffinity()
+}
+
+func getResourceRequirements(cluster *apimgmtv3.Cluster) *corev1.ResourceRequirements {
+	if cluster.Spec.ClusterAgentDeploymentCustomization.OverrideResourceRequirements != nil {
+		return cluster.Spec.ClusterAgentDeploymentCustomization.OverrideResourceRequirements
+	}
+
+	return nil
+}
```

Comment:
These two functions are identical to the ones in this file: https://github.com/rancher/rancher/pull/41203/files#diff-a545c93129913c64774c595c8b394c2a6a406c4894c7cfec4bceb8a7cd9eceabR63